src/theory/evaluator.h

```diff
@@ -88,6 +96,9 @@ class Evaluator
    * `args` and the corresponding values `vals`. The function returns a null
    * node if there is a subterm that is not constant under the substitution or
    * if an operator is not supported by the evaluator.
+   *
+   * Note: The evaluator expects that `n` has been rewritten to the point where
+   * it does not contain non-essential operators (e.g. FLOATINGPOINT_EQ).
    */
```

Comment:
Ah; useful.  Could we Assert that rewriting doesn't change it?

Reply:
Good point, I will add an assert.